feat: add molecules info to python

pilatmartin opened the pull request:
Adds `Molecules::info` to python bindings.

**Usage**
Properties can be accessed directly:
```py
molecules = Molecules("/path/to/file")
info = molecules.info()
print(info.total_atoms) # 29874
```
Or the result can be converted to a `dict` that can be serialized:
```py
info.to_dict()
```

**Example output (receptor.pdb from acc2 examples)**
```json
{
    "total_molecules": 1,
    "total_atoms": 29874,
    "atom_type_counts": [
      {
        "symbol": "N",
        "count": 2386
      },
      {
        "symbol": "C",
        "count": 9738
      },
      {
        "symbol": "O",
        "count": 2723
      },
      {
        "symbol": "H",
        "count": 14950
      },
      {
        "symbol": "S",
        "count": 77
      }
    ]
}
```

krab1k commented:
Can you change that so there is no code duplication? Currently, essentially the same thing is provided by the MoleculeSet::info function. I would suggest the following:
1. Create a common function, MoleculeSet::get_stats(), which would return the struct with the counts.
2. MoleculeSet::info() would use it and just print the results.
3. Python bindings may even return the structure directly.

pilatmartin commented:
> Can you change that so there is no code duplication? Currently, essentially the same thing is provided by the MoleculeSet::info function. I would suggest the following:
> 
>     1. Create a common function, MoleculeSet::get_stats(), which would return the struct with the counts.
> 
>     2. MoleculeSet::info() would use it and just print the results.
> 
>     3. Python bindings may even return the structure directly.

**Changes made**
- renamed `MoleculeInfo` to `MoleculeSetStats`
- moved `MoleculeSetStats` to `molecule_set.h`
- removed `to_dict()` from `MoleculeSetStats` and moved it to pybind11 config
- added `MoleculeSet::get_stats()`
- `MoleculeSet::info()` is using `get_stats()` and prints the result as before
- `Molecules::info()` (in bindings) returns the `MoleculeSetStats`
